flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/SimpleConsumerThread.java

```diff
@@ -370,8 +370,8 @@ else if (partitionsRemoved) {
 								keyPayload.get(keyBytes);
 							}
 
-							final T value = deserializer.deserialize(keyBytes, valueBytes,
-									currentPartition.getTopic(), currentPartition.getPartition(), offset);
+							final T value = deserializer.deserialize(
```

Comment:
Thank you for suggestion, I will try

Reply:
Methods of 0.8 Consumer have different signatures compared to modern Kafka - namely they throw base Exception.

1. Change signature of deserialize(ConsumerRecord) to throw Exception instead of IOException. 
2. Wrap and re-throw in every deserialize(ConsumerRecord)  implementation which is 0.8 or for Kafka-base (KeyedDeserializationSchemaWrapper, JSONKeyValueDeserializationSchema, NonContinousOffsetsDeserializationSchema and TypeInformationKeyValueSerializationSchema) 

@azagrebin, @stevenzwu which approach do you prefer?

Reply:
Done in 2adbacb